src/dbus.c

```diff
@@ -333,6 +333,9 @@ static void on_get_server_information(GDBusConnection *connection,
 
 void notification_closed(notification *n, enum reason reason)
 {
+        if (n->redisplayed)
```

Comment:
Good catch. But I'd like to have this separated from DBus. I think we should fix it in queues_notification_close.